screens/Home/HomeScreen.tsx

```diff
@@ -11,153 +11,98 @@ import { useSafeAreaInsets } from "react-native-safe-area-context"
+    const events: Event[] = isFuture ? futureEvents : pastEvents
+    return events.map((event: Event) => (
+      <TouchableOpacity key={event.title}>
+        <EventCard event={event} />
+      </TouchableOpacity>
+    ))
   }
-  // Can be replaced by EventCardProps later
-  const renderEvents = (event: {
-    title: string
-    location: string
-    latitude: number
-    longitude: number
-    description: string
-    date: string
-    imageUrl: string
-  }) => (
-    <TouchableOpacity key={event.title}>
-      <EventCard
-        title={event.title}
-        location={event.location}
-        latitude={event.latitude}
-        longitude={event.longitude}
-        description={event.description}
-        date={event.date}
-        imageUrl={event.imageUrl}
-        // Should redirect to an event dedicated page where the user can register for the event
-      />
-    </TouchableOpacity>
-  )
 
   return (
     //<TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
-      <View
-        style={[
-          styles.view,
-          { paddingBottom: insets.bottom, paddingTop: insets.top },
-        ]}
-      >
-        {/* Add the choose tab component from Gael pull request */}
-        
-        <View style={styles.searchAndMap}>
-          <View>
-            <TextInput
-              placeholder="Search..."
-              style={[styles.input, globalStyles.text]}
-              onChangeText={handleSearch}
-            />
-          </View>
-          {/* Need to style the opacity to render it smaller*/}
-          <TouchableOpacity
-            style={styles.map}
-            onPress={() =>
-              navigation.navigate("EventMap", { events })
-            }
-          >
-            <Text> Show on the Map</Text>
-          </TouchableOpacity>
-        </View>
+    <View
+      style={[
+        styles.view,
+        { paddingBottom: insets.bottom, paddingTop: insets.top },
+      ]}
+    >
+      {/* Add the choose tab component from Gael pull request */}
 
-        <View style={styles.containerEvent}>
-          <Text style={styles.title}>Futur Events</Text>
-          <ScrollView style={styles.eventList}>
-            {/* Add the event card componrent */}
-            {futureEvents.map((event) => renderEvents(event))}
-            <Text style={styles.title}>Past Events </Text>
-            {/* Add the events card component Need a way to differentiate from past and futur event */}
-            {pastEvents.map((event) => renderEvents(event))}
-          </ScrollView>
+      <View style={styles.searchAndMap}>
+        <View>
+          <TextInput
+            placeholder="Search..."
+            style={[styles.input, globalStyles.text]}
+            onChangeText={handleSearch}
+          />
         </View>
+        {/* Need to style the opacity to render it smaller*/}
+        <TouchableOpacity
+          style={styles.map}
+          onPress={() =>
+            navigation.navigate("EventMap",  {events: futureEvents})
```

Comment:
I know you didn't introduce the error but if you can fix it...?

Reply:
Which error?